[12.0] [FIX] [hr_employee_service_contract]

YuriyGural opened the pull request:
Сorrect filling of the fields "service_start_date", "service_termination_date".
It depends on the status change from "New" to "Running", and also if the "End date" field in one of the contracts is empty.

YuriyGural committed:
[12.0] [FIX] [hr_employee_service]

Сorrect filling of the fields "service_start_date", "service_termination_date".
It depends on the status change from "New" to "Running", and also if the "End date" field in one of the contracts is empty.

etobella commented:
Tests are failing due to your change, also, why did you change the function?

alexey-pelykh commented on hr_employee_service_contract/models/hr_employee.py:
```diff
@@ -42,22 +42,31 @@ def _compute_first_contract_id(self):
         pass
 
+    @api.multi
+    def _get_last_contract_id(self, employee_id):
```
@YuriyGural if you're changing the behaviour, please describe the change as well as provide tests for it

YuriyGural replied:
```
    def test_6(self):
        employee = self.SudoEmployee.create({
            'name': 'Employee #6',
        })
        contract_1 = self.Contract.create({
            'name': 'Employee #6 Contract #1',
            'employee_id': employee.id,
            'wage': 5000.0,
            'state': 'open',
            'date_start': self.today - relativedelta(years=5),
            'date_end': self.today - relativedelta(years=1),
        })
        contract_2 = self.Contract.create({
            'name': 'Employee #6 Contract #2',
            'employee_id': employee.id,
            'wage': 6000.0,
            'state': 'draft',
            'date_start': self.today - relativedelta(years=2),
            'date_end': False,
        })

        self.assertEqual(
            employee.service_termination_date,
            self.today - relativedelta(years=1)
        )

        contract_2.write({'state': 'open'})
        self.assertEqual(
            employee.service_termination_date,
            False
        )
```

alexey-pelykh replied:
https://github.com/OCA/hr/blob/12.0/hr_employee_service/tests/test_hr_employee_service.py

YuriyGural committed:
Update hr_employee.py

YuriyGural committed:
Update hr_employee.py

YuriyGural committed:
Update hr_employee.py

It was important for me to fix the situation when we have several contracts without an end date (in my case, these were wage changes) and in the last contract I put the end date of the contract in connection with the dismissal of an employee.
